src/canvas/widgets/network_graph.rs

```diff
@@ -190,29 +435,138 @@ impl NetworkGraphWidget for Painter {
+
+                    if let (Some(outside_rx_point), Some(inside_rx_point)) =
+                        (outside_rx_point, inside_rx_point)
+                    {
+                        let old = *outside_rx_point;
+
+                        let new_point = (
+                            time_start,
+                            interpolate_points(outside_rx_point, inside_rx_point, time_start),
+                        );
+
+                        // debug!(
+                        //     "Interpolated between {:?} and {:?}, got rx for time {:?}: {:?}",
+                        //     outside_rx_point, inside_rx_point, time_start, new_point
+                        // );
+
+                        if let Some(to_replace) = network_data_rx.get_mut(rx_start_pos) {
+                            *to_replace = new_point;
+                            Some((rx_start_pos, old))
+                        } else {
+                            None // Failed to get mutable reference.
+                        }
+                    } else {
+                        None // Point somehow doesn't exist in our network_data_rx
+                    }
+                } else {
+                    None // Point is already "leftmost", no need to interpolate.
+                }
+            } else {
+                None // There is no point.
+            };
+
+            let interpolated_tx_point = if let Some(tx_end_pos) = network_data_tx
+                .iter()
+                .position(|(time, _data)| *time >= time_start)
+            {
+                if tx_end_pos > 1 {
+                    let tx_start_pos = tx_end_pos - 1;
+                    let outside_tx_point = network_data_tx.get(tx_start_pos);
+                    let inside_tx_point = network_data_tx.get(tx_end_pos);
+
+                    if let (Some(outside_tx_point), Some(inside_tx_point)) =
+                        (outside_tx_point, inside_tx_point)
+                    {
+                        let old = *outside_tx_point;
+
+                        let new_point = (
+                            time_start,
+                            interpolate_points(outside_tx_point, inside_tx_point, time_start),
+                        );
+
+                        if let Some(to_replace) = network_data_tx.get_mut(tx_start_pos) {
+                            *to_replace = new_point;
+                            Some((tx_start_pos, old))
+                        } else {
+                            None // Failed to get mutable reference.
+                        }
+                    } else {
+                        None // Point somehow doesn't exist in our network_data_tx
+                    }
+                } else {
+                    None // Point is already "leftmost", no need to interpolate.
+                }
+            } else {
+                None // There is no point.
+            };
+
+            // TODO: Cache network results: Only update if:
```

Comment:
This is for the future (i.e. after this PR).  Ideally we can re-use some max entry calculations.